### PR TITLE
Get api with profileId that created it or with other's profileId 

### PR DESCRIPTION
**Pull request by danifara3**
Added logic for getting partialApi details
added a get-api.dto file

added a build command to the prod:start script so that it builds before starting the server

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "watch:dev": "nest start --watch",
     "watch:listen": "nest start --watch --config listener.json",
     "start:debug": "nest start --debug --watch",
+    "prestart:prod": "npm run build",
```
Comment (Chimexx) [inline package.json:19]: Ahh, Oga what is this for?

File: src/api/api.service.ts [hunk 2]
```diff
@@ -67,15 +69,28 @@ export class ApiService {
...
+        return api;
+      } else {
+        const {
+          ['base_url']: unused1,
+          ['visibility']: unused2,
+          ['subscriptions']: unused3,
+          ['status']: unused4,
+          ['secretKey']: unused5,
+          ...partialApi
+        } = api;
```
Comment (Chimexx) [inline src/api/api.service.ts:82]: Frankly speaking, I don't know what you are doing

File: src/api/dto/get-api.dto.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsArray, IsDate, IsEnum, IsNumber, IsString } from 'class-validator';
+import { Status } from 'src/common/enums/apiVerification.enum';
+import { Visibility } from 'src/common/enums/visibility.enum';
+export class GetApiDto {
+  @ApiProperty()
+  @IsString()
+  id: string;
+
+  @ApiProperty()
```
Comment (Chimexx) [inline src/api/dto/get-api.dto.ts:10]: You don't need dto to query a db

File: src/entities/api.entity.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ export class Api extends SharedEntity {
   description: string;
 
   @Column()
-  base_url: string;
+  base_url?: string;
```
Comment (Chimexx) [inline src/entities/api.entity.ts:24]: You modified entity? no nah

**Review by Chimexx — CHANGES_REQUESTED** [review]
This is not what you are supposed to do... The dummy repo is there for your reference.

File: package.json (hunk 1 above)
Comment (danifara3) [reply package.json:19]: builds before starting, changes don't reflect unless you do that first

File: src/entities/api.entity.ts (hunk 4 above)
Comment (danifara3) [reply src/entities/api.entity.ts:24]: my bad, changed

File: src/api/api.service.ts (hunk 2 above)
Comment (danifara3) [reply src/api/api.service.ts:82]: removing data not needed if not profileId

File: src/api/dto/get-api.dto.ts (hunk 3 above)
Comment (danifara3) [reply src/api/dto/get-api.dto.ts:10]: if i add :Promise<Api> i get error for my return statement ( return partialApi;), should i remove the promise type all together?

File: src/entities/api.entity.ts (hunk 4 above)
Comment (PhilemonBrain) [reply src/entities/api.entity.ts:24]: @danifara3 these fields u made optionally are actually required

File: src/entities/api.entity.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export class Api extends SharedEntity {
   term_of_use: string;
 
   @Column('text', { array: true, nullable: true, default: [] })
-  subscriptions: string[];
+  subscriptions?: string[];
```
Comment (PhilemonBrain) [inline src/entities/api.entity.ts:42]: @danifara3 please why is this optional?

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**